app/questionnaire/router.py

```diff
@@ -202,23 +277,21 @@ def _get_allowable_path(self, routing_path):
         return allowable_path
 
     def _get_incomplete_section_keys(self):
-        all_section_ids = self._schema.get_section_ids()
-
-        all_section_keys = []
-        for section_id in all_section_ids:
+        enabled_section_keys = []
+        for section_id in self.enabled_section_ids:
             repeating_list = self._schema.get_repeating_list_for_section(section_id)
 
             if repeating_list:
                 for list_item_id in self._list_store[repeating_list].items:
                     section_key = (section_id, list_item_id)
-                    all_section_keys.append(section_key)
+                    enabled_section_keys.append(section_key)
             else:
                 section_key = (section_id, None)
-                all_section_keys.append(section_key)
+                enabled_section_keys.append(section_key)
 
         incomplete_section_keys = [
             (section_id, list_item_id)
-            for section_id, list_item_id in all_section_keys
+            for section_id, list_item_id in enabled_section_keys
```

Comment:
Possibly not for this pull request, but this should be a `_get_first_incomplete_section_key` method that returns as soon as it finds an incomplete section - we only care about the next incomplete section or if the survey is complete. Thoughts?

Reply:
I think if we leave this then it probably won't be looked at in a long time so I've decided to refactor into `_get_first_incomplete_section_key`.